evm_arithmetization/src/generation/mpt.rs

```diff
@@ -481,15 +466,21 @@ where
                     .map_err(|_| ProgramError::IntegerTooLarge)?,
             ));
             // Write `value_ptr_ptr`.
-            storage_leaves.push(Some((trie_data.len()).into()));
+            let leaves = parse_value(value)?
+                .into_iter()
+                .map(Some)
+                .collect::<Vec<_>>();
+            let leaf = match leaves.len() {
+                0 => Some(0.into()),
```

Comment:
We don't store 0s in storage slots. Trying to insert a 0 will delete it instead. I don't think we should be pushing a 0 value here if the `leaves` len is null?

Reply:
This is the value that is stored in the linked lists... Since we are storing the values directly and not the pointers anymore, then if there is no storage, we store 0 to indicate that we point to the empty pointer (when returning the value in the Kernel, we first check whether the value is 0)

Reply:
But this is called when parsing the trie at the beginning. We shouldn't encounter any 0s there anyway, should we ? 🤔